worker/uniter/relation/resolver.go

```diff
@@ -176,12 +176,28 @@ func (r *relationsResolver) nextHookForRelation(localStateDir *StateDir, remote
 			if err != nil {
 				return hook.Info{}, errors.Trace(err)
 			}
+
+			// Consult the life of the local unit and/or app to
+			// figure out if its the local or the remote unit going
+			// away. Note that if the app is removed, the unit will
+			// still be alive but its parent app will by dying.
+			localUnitLife, localAppLife, err := r.stateTracker.LocalUnitAndApplicationLife()
+			if err != nil {
+				return hook.Info{}, errors.Trace(err)
+			}
+
+			var departee = unitName
+			if localUnitLife != life.Alive || localAppLife != life.Alive {
+				departee = r.stateTracker.LocalUnitName()
```

Comment:
so I see that you set 'departee' to the local unit, but I don't see where you set it to the remote unit.
Am I missing an else clause somewhere?

Reply:
L189 inits the var with the _remote_ unit's name. Given (AFAICT from the code) that we do not have permissions to query the life for the remote unit/app, we instead query the life for the local unit/app and override the departee to the local unit on L191.

Reply:
yeah, I just misread 189. Thanks for clarifying.